db/flush_job_test.cc

```diff
@@ -487,6 +507,137 @@ TEST_F(FlushJobTest, Snapshots) {
   job_context.Clean();
 }
 
+class FlushJobTimestampTest : public FlushJobTestBase {
+ public:
+  FlushJobTimestampTest()
+      : FlushJobTestBase(test::PerThreadDBPath("flush_job_ts_gc_test"),
+                         test::ComparatorWithU64Ts()) {}
+
+  void AddKeyValueToMemtable(MemTable* memtable, Slice key, uint64_t ts,
+                             SequenceNumber seq, ValueType value_type,
+                             Slice value) {
+    std::string key_str(key.data(), key.size());
+    std::string ts_str;
+    PutFixed64(&ts_str, ts);
+    key_str.append(ts_str);
+    memtable->Add(seq, value_type, key_str, value);
+  }
+
+ protected:
+  static constexpr uint64_t kStartTs = 10;
+  static constexpr SequenceNumber kStartSeq = 0;
+  SequenceNumber curr_seq_{kStartSeq};
+  std::atomic<uint64_t> curr_ts_{kStartTs};
+};
+
+TEST_F(FlushJobTimestampTest, AllKeysExpired) {
+  ColumnFamilyData* cfd = versions_->GetColumnFamilySet()->GetDefault();
+  autovector<MemTable*> to_delete;
+
+  {
+    MemTable* new_mem = cfd->ConstructNewMemtable(
+        *cfd->GetLatestMutableCFOptions(), kMaxSequenceNumber);
+    new_mem->Ref();
```

Comment:
Do we need to `Unref` / `delete` this?

Reply:
We do not have to do it here. `MemTableListVersion::Remove()` unrefs the memtable, and `JocContext::Clean()` deletes it.